VIC_RBM/model_run/result_analysis/scripts/my_functions.py

```diff
@@ -265,17 +265,27 @@ def calc_monthly_data(data):
 #==============================================================
 #==============================================================
 
+def wateryear(calendar_date):
+	if calendar_date.month >= 10:
+		return calendar_date.year+1
+	return calendar_date.year
+
 def calc_ts_stats_by_group(data, by, stat):
 	'''This function calculates statistics of time series data grouped by year, month, etc
 
 	Input:
 		df: a [pd.DataFrame/Series] object, with index of time
-		by: string of group by, (select from 'year' or 'month')
+		by: string of group by, (select from 'year' or 'month' or 'WY')
 		stat: statistics to be calculated, (select from 'mean')
 		(e.g., if want to calculate monthly mean seasonality (12 values), by='month' and stat='mean')
 
 	Return:
 		A [dateframe/Series] object, with group as index (e.g. 1-12 for 'month')
+
+	Require:
```

Comment:
I like the require piece- it makes it clearer for a user as to what they need to put in and keys us in to what's important.